tests/conftest.py

```diff
@@ -197,3 +213,65 @@ def cylc_validate_cli(capsys, caplog):
 @pytest.fixture(scope='module')
 def mod_cylc_validate_cli(mod_capsys, mod_caplog):
     return _cylc_validate_cli(mod_capsys, mod_caplog)
+
+
+@pytest.fixture
+def file_poll():
+    """Poll for the existance of a file.
+    """
+    def _inner(
+        fpath: "Path", timeout: int = 5, inverse: bool = False
+    ):
+        if inverse:
+            def check(f):
+                not f.exists()
+        else:
+            def check(f):
+                f.exists()
+
+        timeout_func(
+            partial(check, fpath),
+            f'file {fpath} not found after {timeout} seconds'
+        )
```

Comment:
FWIW, if there are meant to be tests relying on this
```suggestion
        timeout_func(
            lambda: fpath.exists() != inverse,
            f"file {fpath} {'still' if inverse else 'not'} found after "
            f"{timeout} seconds",
            timeout
        )
```